### PR TITLE
Basic UI

### PR DESCRIPTION
**Pull request by asdine**
This PR adds a UI to Regula.
The UI is built using VueJS and Material design.
List of changes:
- New `mock` package: moved api unexported mocks to a shared package
- New `ui` package: Contains http handlers serving the UI SPA
- New `ui/app` directory: Contains the UI SPA source code
  - It displays a simple sidebar for now with the tree of rulesets
- Travis now runs the UI tests
- package `store`: Removed unnecessary `Latest` and `OneByVersion` methods

regula-ui

### CODE REVIEW

File: cmd/regula/cli/cli.go [hunk 1]
```diff
@@ -25,8 +25,9 @@ import (
 // Config holds the server configuration.
 type Config struct {
 	Etcd struct {
-		Endpoints []string `config:"etcd-endpoints"`
-		Namespace string   `config:"etcd-namespace,required"`
+		Endpoints   []string      `config:"etcd-endpoints"`
+		Namespace   string        `config:"etcd-namespace,required"`
+		DialTimeout time.Duration `config:"etcd-dial-timemout"`
```
Comment (tealeg) [inline cmd/regula/cli/cli.go]: Is this change strictly necessary / related?

File: ui/app/src/components/Sidebar.vue [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  const tree = {};
+
+  // turns each ruleset path in the form a/b/c into a
+  // temporary tree of this form:
+  // {a: {b: {c: []}}} and fills the tree variable
+  rulesets.forEach(({ path }) => {
+    let node = tree;
+
+    path.split('/').forEach((chunk, idx, list) => {
+      if (!Object.prototype.hasOwnProperty.call(node, chunk)) {
```
Comment (tealeg) [inline ui/app/src/components/Sidebar.vue]: To us civilians, this line here is worthy of a comment.

File: ui/app/src/components/Sidebar.vue [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    items: [],
+  }),
+
+  mounted() {
+    this.fetchRulesets();
+  },
+
+  methods: {
+    fetchRulesets() {
+      fetch('i/rulesets/')
```
Comment (tealeg) [inline ui/app/src/components/Sidebar.vue]: It almost make me just want threads ;-)

File: ui/handler.go [hunk 4]
```diff
@@ -1,20 +1,109 @@
...
+
+	type response struct {
+		Rulesets []ruleset `json:"rulesets"`
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var resp response
+		var token string
+
+		// simulate a Do-While loop in Go
```
Comment (tealeg) [inline ui/handler.go]: :tulip: I think it would be more useful to indicate in your comment that you want this loop to run at least once, no matter the value of token.  Referencing a flow control construct from other languages is requiring a bit too much meta-knowledge for my tastes ;-)

File: cmd/regula/cli/cli.go (hunk 1 above)
Comment (asdine) [reply cmd/regula/cli/cli.go]: It is not related at all actually, it was just in my way, to be honest. About the necessity of this change, I thought it would be useful to be able to control how long someone accepts to wait for etcd to be ready I can remove it though, WDYT?

File: ui/app/.browserslistrc [hunk 5]
```diff
@@ -0,0 +1,3 @@
+> 1%
+last 2 versions
+not ie <= 8
```
Comment (YassineE) [inline ui/app/.browserslistrc]: The famous IE8 😂 
I have very bad memories with it.

File: ui/app/package.json [hunk 6]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "regula",
+  "version": "0.6.0",
```
Comment (YassineE) [inline ui/app/package.json]: ❓ Why do we are sync between the backend version and the UI one?

Comment (YassineE) [reply ui/app/package.json]: It should be `0.1.0`, isn't it?

Comment (asdine) [reply ui/app/package.json]: It depends, they are part of the same project so I think that we want both projects to be synchronized, at least for now
